### PR TITLE
Added StdDev accumulator builders

### PR DESCRIPTION
**Pull request by rozza**
JAVA-1935

### CODE REVIEW

File: driver-core/src/main/com/mongodb/client/model/Accumulators.java [hunk 1]
```diff
@@ -137,6 +137,43 @@
         return accumulator("$addToSet", fieldName, expression);
     }
 
+    /**
+     * Gets a field name for a $group operation representing the sample standard deviation of the values of the given expression
+     * when applied to all members of the group.
+     *
+     * <p>Use if the values encompasses the entire population of data you want to represent and do not wish to generalize about
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/client/model/Accumulators.java]: if the values encompass

File: driver-core/src/main/com/mongodb/client/model/Accumulators.java [hunk 2]
```diff
@@ -137,6 +137,43 @@
...
+     */
+    public static <TExpression> BsonField stdDevPop(final String fieldName, final TExpression expression) {
+        return accumulator("$stdDevPop", fieldName, expression);
+    }
+
+    /**
+     * Gets a field name for a $group operation representing the sample standard deviation of the values of the given expression
+     * when applied to all members of the group.
+     *
+     * <p>Use if the values encompasses a sample of a population of data from which to generalize about the population.</p>
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/client/model/Accumulators.java]: if the values encompass

File: driver-core/src/test/unit/com/mongodb/client/model/AggregatesSpecification.groovy [hunk 3]
```diff
@@ -114,17 +118,10 @@ class AggregatesSpecification extends Specification {
                      first('first', '$quantity'),
                      last('last', '$quantity'),
                      push('all', '$quantity'),
-                     addToSet('unique', '$quantity'))) == groupDocument
-
-        toBson(group(null,
```
Comment (jyemin) [inline driver-core/src/test/unit/com/mongodb/client/model/AggregatesSpecification.groovy]: This is testing the overloaded group method that takes a List<BsonField>, so it shouldn't be removed

**Comment by jyemin** [comment]
Are you planning to do the reference docs all at once for all agg-related changes, or do you want to code review those one at a time?

File: driver-core/src/main/com/mongodb/client/model/Accumulators.java (hunk 1 above)
Comment (rozza) [inline driver-core/src/main/com/mongodb/client/model/Accumulators.java]: Sorry copy pasta from the 3.1-dev release notes - updated

File: driver-core/src/main/com/mongodb/client/model/Accumulators.java (hunk 2 above)
Comment (rozza) [inline driver-core/src/main/com/mongodb/client/model/Accumulators.java]: Sorry copy pasta from the 3.1-dev release notes - updated

File: driver-core/src/test/unit/com/mongodb/client/model/AggregatesSpecification.groovy (hunk 3 above)
Comment (rozza) [inline driver-core/src/test/unit/com/mongodb/client/model/AggregatesSpecification.groovy]: Doh! sorry, somehow I missed it.

**Comment by rozza** [comment]
Not sure what Doc updates would be required - as we just link to the API docs.  See: http://mongodb.github.io/mongo-java-driver/3.1/builders/aggregation/#group

**Comment by jyemin** [comment]
I didn't realize that not all the Accumulators were documented in the reference docs.  So yes, nothing to do here.

File: driver-core/src/test/unit/com/mongodb/client/model/AggregatesSpecification.groovy [hunk 4]
```diff
@@ -114,17 +118,23 @@ class AggregatesSpecification extends Specification {
...
-                     [sum('sum', parse('{ $multiply: [ "$price", "$quantity" ] }')),
-                      avg('avg', '$quantity'),
-                      min('min', '$quantity'),
-                      max('max', '$quantity'),
-                      first('first', '$quantity'),
-                      last('last', '$quantity'),
-                      push('all', '$quantity'),
-                      addToSet('unique', '$quantity')])) == groupDocument
+                     addToSet('unique', '$quantity'),
```
Comment (jyemin) [inline driver-core/src/test/unit/com/mongodb/client/model/AggregatesSpecification.groovy]: Are these no longer lined up properly?

**Comment by jyemin** [comment]
lgtm

**Comment by rozza** [comment]
Merged into my local 3.2 branch

**Pull request closed without merging**